src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLBlitLoops.m

```diff
@@ -0,0 +1,791 @@
+                (height == (dstInfo.bounds.y2 - dstInfo.bounds.y1))) {
+                // mtlbuf.contents have same dimensions as of pDst
+                copyFromMTLBuffer(pDst, mtlbuf, 0, byteLength, convertFromPre);
+            } else {
+                // mtlbuf.contents have smaller dimensions than pDst
+                // copy each row from mtlbuf.contents at appropriate position in pDst
+                // Note : pDst is already addjusted for offsets using PtrAddBytes above
+
+                int rowSize = width * dstInfo.pixelStride;
+                for (int y = 0; y < height; y++) {
+                    copyFromMTLBuffer(pDst, mtlbuf, y * rowSize, rowSize, convertFromPre);
+                    pDst = PtrAddBytes(pDst, dstInfo.scanStride);
+                }
+            }
+
+#ifndef USE_STATIC_BUFFER
+            [mtlbuf release];
+#endif // USE_STATIC_BUFFER
+        }
+        SurfaceData_InvokeRelease(env, dstOps, &dstInfo);
+    }
+    SurfaceData_InvokeUnlock(env, dstOps, &dstInfo);
+}
+
+void
+MTLBlitLoops_CopyArea(JNIEnv *env,
+                      MTLContext *mtlc, BMTLSDOps *dstOps,
+                      jint x, jint y, jint width, jint height,
+                      jint dx, jint dy)
+{
+#ifdef DEBUG
+    J2dTraceImpl(J2D_TRACE_VERBOSE, JNI_TRUE, "MTLBlitLoops_CopyArea: bdst=%p [tex=%p] %dx%d | src (%d, %d), %dx%d -> dst (%d, %d)",
+            dstOps, dstOps->pTexture, ((id<MTLTexture>)dstOps->pTexture).width, ((id<MTLTexture>)dstOps->pTexture).height, x, y, width, height, dx, dy);
+#endif //DEBUG
+
+    @autoreleasepool {
+    id<MTLCommandBuffer> cb = [mtlc createCommandBuffer];
+    id<MTLBlitCommandEncoder> blitEncoder = [cb blitCommandEncoder];
+
+    // Create an intrermediate buffer
+    int totalBuffsize = width * height * 4;
+    id <MTLBuffer> buff = [[mtlc.device newBufferWithLength:totalBuffsize options:MTLResourceStorageModePrivate] autorelease];
+
+    [blitEncoder copyFromTexture:dstOps->pTexture
+            sourceSlice:0 sourceLevel:0 sourceOrigin:MTLOriginMake(x, y, 0) sourceSize:MTLSizeMake(width, height, 1)
+             toBuffer:buff destinationOffset:0 destinationBytesPerRow:(width * 4) destinationBytesPerImage:totalBuffsize];
+
+    [blitEncoder copyFromBuffer:buff
+            sourceOffset:0 sourceBytesPerRow:width*4 sourceBytesPerImage:totalBuffsize sourceSize:MTLSizeMake(width, height, 1)
+            toTexture:dstOps->pTexture destinationSlice:0 destinationLevel:0 destinationOrigin:MTLOriginMake(x + dx, y + dy, 0)];
+    [blitEncoder endEncoding];
+
+    [cb commit];
+    //[cb waitUntilCompleted];
+
+    /*[blitEncoder
+            copyFromTexture:dstOps->pTexture
+            sourceSlice:0 sourceLevel:0 sourceOrigin:MTLOriginMake(x, y, 0) sourceSize:MTLSizeMake(width, height, 1)
+            toTexture:dstOps->pTexture destinationSlice:0 destinationLevel:0 destinationOrigin:MTLOriginMake(x + dx, y + dy, 0)];
+    [blitEncoder endEncoding];*/
+
+    }
+    // TODO:
```

Comment:
The comment is not needed or this is incomplete?

Reply:
@aghaisas Do we need this code and comment. i remember we discussed about this intersect logic sometime back and you tested and mentioned that calculating intersects is not having any effect.

Reply:
The commented code is not needed. I will remove it.
Regarding TODO comment, I will investigate under JDK-8256029.